kernel/src/sched/sched_class/fair.rs

```diff
@@ -96,27 +100,66 @@ pub const fn nice_to_weight(nice: Nice) -> u64 {
 #[derive(Debug)]
 pub struct FairAttr {
     weight: AtomicU64,
+    pending_weight: AtomicU64,
```

Comment:
Why was `pending_weight` introduced here for delayed weight updates? I'm not sure if this change is appropriate. Could you add some comments or a reference to explain this behavior?